camelot/core/files/storage.py

```diff
@@ -51,24 +51,25 @@ class StoredFile:
     its methods should block.
     """
 
-    def __init__(self, storage: 'Storage', name: str):
+    def __init__(self, storage: 'Storage', name: PurePosixPath):
         """
         :param storage: the storage in which the file is stored
         :param name: the key by which the file is known in the storage
         """
+        assert isinstance(name, PurePosixPath)
         self.storage = storage
-        self.name = name
+        self.name: PurePosixPath = name
 
     @property
     def verbose_name(self) -> str:
         """The name of the file, as it is to be displayed in the GUI"""
-        return self.name
+        return self.name.as_posix()
```

Comment:
opgepast : verbose_name is 'displayed in GUI', dus daar mag in wezen geen enkele logica mee of op gebeuren ...

Reply:
Wil je dan dat ik de verbose name ervoor 'bereken'? en wat met paden in de name moeten die erbij of enkel de filename?

Reply:
verbose_name mag 'berekend' worden, hier in de property, of op voorhand, dat maakt nt echt veel uit.

maar die mag nergens gebruikt worden in logica ...